app/Http/Controllers/NicknameController.php

```diff
@@ -14,6 +18,102 @@ public function __construct(ResponseInterface $resProvider)
     {
         $this->resProvider = $resProvider;
     }
+
+    /**
+     * @OA\Get(path="/add_nick_name",
+     *   tags={"nickname"},
+     *   summary="Add New nick name",
+     *   description="",
+     *   operationId="addNickName",
+     *   @OA\Parameter(
+     *     name="nick_name",
+     *     required=true,
+     *     in="query",
+     *     description="Unique nickname required with max 50 characters",
+     *     @OA\Schema(
+     *         type="string"
+     *     )
+     *   ),
+     *   @OA\Parameter(
+     *     name="visibility_status",
+     *     in="query",
+     *     @OA\Schema(
+     *         type="integer",
+     *     ),
+     *     description="Visibility status required",
+     *   ),
+     *   
+     *   @OA\Response(response=400, description="Something went wrong")
+     *   @OA\Response(response=200, description="success",  @OA\Schema(ref="#/components/schemas/Nickname"))
+     * )
+     */
+    public function addNickName(AddNickNameRequest $request)
+    {
+        $user = $request->user();
+        try {
+            $nickname = Nickname::createNickname($user->id, $request->all());
+            return $this->resProvider->apiJsonResponse(200, 'Success', $nickname, '');
+        } catch (\Throwable $e) {
+            return $this->resProvider->apiJsonResponse(400, "Something went wrong", '', $e->getMessage());
+        }
+
+    }
+
+
+    /**
+     * @OA\Get(path="/update_nick_name",
+     *   tags={"nickname"},
+     *   summary="update nick name visibility status",
+     *   description="",
+     *   operationId="updatenickName", 
+     *   @OA\Parameter(
+     *     name="visibility_status",
+     *     in="query",
+     *     @OA\Schema(
+     *         type="integer",
+     *     ),
+     *     description="Visibility status required",
+     *   ),
+     *   
+     *   @OA\Response(response=400, description="Something went wrong")
+     *   @OA\Response(response=200, description="Update successfully",  @OA\Schema(ref="#/components/schemas/Nickname"))
+     * )
+     */
+    public function updateNickName($id, UpdateNickNameRequest $request){
+        $user = $request->user();
+        try {
+            $nickname = Nickname::findOrFail($id);
+            $nickname->private = $request->visibility_status;
+            $nickname->update();
+            
+            return $this->resProvider->apiJsonResponse(200, 'Nick name visibility status updated successfully.', $nickname, '');
```

Comment:
Please add nick_name under the message.php config and replace the 'Nick name visibility status updated successfully' message